### PR TITLE
Standardize multiplication involving string column

### PR DESCRIPTION
**Pull request by xinrong-databricks**
## Proposal
Multiplication involves a string column can only be applied to an int column or literal; otherwise, a TypeError should be raised.

### CODE REVIEW

File: databricks/koalas/base.py [hunk 1]
```diff
@@ -245,11 +245,17 @@ def __mul__(self, other):
...
-            (isinstance(other, IndexOpsMixin) and isinstance(other.spark.data_type, IntegralType))
-            or isinstance(other, int)
-        ):
-            return column_op(SF.repeat)(self, other)
+        if isinstance(self.spark.data_type, StringType):
+            if (
+                isinstance(other, IndexOpsMixin)
+                and isinstance(other.spark.data_type, IntegralType)
+                or isinstance(other, int)
+            ):
```
Comment (ueshin) [inline databricks/koalas/base.py:253]: Shall we make sure the range of `and` and `or` when they are mixed?

```suggestion
            if (
                isinstance(other, IndexOpsMixin) and isinstance(other.spark.data_type, IntegralType)
            ) or isinstance(other, int):
```

**Review by ueshin — COMMENTED** [review]
Otherwise, LGTM, pending tests.

File: databricks/koalas/base.py (hunk 1 above)
Comment (xinrong-databricks) [reply databricks/koalas/base.py]: Makes sense, this way makes code clearer! Updated.

**Review by ueshin — APPROVED** [review]

**Comment by xinrong-databricks** [comment]
@ueshin Thank you! Merged.

**Pull request merged**